pkg/kbagent/service/command.go

```diff
@@ -186,63 +160,17 @@ func runCommandX(ctx context.Context, action *proto.ExecAction, parameters map[s
 			return
 		}
 
-		var wg sync.WaitGroup
-		wg.Add(3)
-
-		var ioCopyError error
-		go func() {
-			defer wg.Done()
-			if stdinReader != nil {
-				defer stdin.Close()
-				_, copyErr := io.Copy(stdin, stdinReader)
-				if copyErr != nil {
-					if errors.Is(copyErr, os.ErrClosed) {
-						return
-					}
-					ioCopyError = errors.Wrapf(proto.ErrFailed, "failed to copy from input reader to stdin: %v", copyErr)
-				}
-			}
-		}()
-		go func() {
-			defer wg.Done()
-			if stdoutWriter != nil {
-				_, copyErr := io.Copy(stdoutWriter, stdout)
-				if copyErr != nil {
-					if errors.Is(copyErr, os.ErrClosed) {
-						return
-					}
-					ioCopyError = errors.Wrapf(proto.ErrFailed, "failed to copy stdout to output writer: %v", copyErr)
-				}
-			}
-		}()
-		go func() {
-			defer wg.Done()
-			if stderrWriter != nil {
-				_, copyErr := io.Copy(stderrWriter, stderr)
-				if copyErr != nil {
-					if errors.Is(copyErr, os.ErrClosed) {
-						return
-					}
-					ioCopyError = errors.Wrapf(proto.ErrFailed, "failed to copy stderr to error writer: %v", copyErr)
-				}
-			}
-		}()
-
-		// wait for the command to finish and the pipes to be closed
 		execErr := cmd.Wait()
 		if execErr != nil {
 			if errors.Is(ctx.Err(), context.DeadlineExceeded) {
 				execErr = proto.ErrTimedOut
 			}
 		}
 
-		// and then wait for the io copy goroutines to finish
-		wg.Wait()
-
 		if execErr != nil {
```

Comment:
errChan <- execErr

Reply:
Solved. Due to branch naming issues, this PR will be closed. The new PR has the same content as this one.
https://github.com/apecloud/kubeblocks/pull/8168